src/ucp/core/ucp_context.c

```diff
@@ -1457,8 +1622,20 @@ static ucs_status_t ucp_fill_config(ucp_context_h context,
         goto err_free_alloc_methods;
     }
 
+    ucs_list_for_each(key_val, &config->list, list) {
+        status = ucp_cached_key_alloc(&key_val_clone,
+                                      key_val->key, key_val->val);
+        if (status != UCS_OK) {
+            status = UCS_ERR_NO_MEMORY;
+            goto err_free_key_list;
+        }
+        ucs_list_add_tail(&context->list, &key_val_clone->list);
```

Comment:
maybe move ucs_list_add_tail inside ucp_cached_key_alloc and rename it to "ucp_config_cached_key_add()"?
it's also used same way from another place

Reply:
Done. I changed the code according to this comment.